dnf/base.py

```diff
@@ -261,10 +266,18 @@ def close(self):
             return
         logger.log(dnf.logging.DDEBUG, 'Cleaning up.')
         self._closed = True
-
-        if (not self.conf.keepcache and
-                not self.ts.isTsFlagSet(rpm.RPMTRANS_FLAG_TEST)):
-            self.clean_used_packages()
+        self.pkgs_to_remove = dnf.persistor.TempfilePersistor(
+            self.conf.cachedir)
+
+        if self.succ_rpm_trans:
+            # delete all packages from last unsuccessful transactions
+            self._tempfiles.update(self.pkgs_to_remove.get_saved_files())
+            self.pkgs_to_remove.empty()
+        if not self.conf.keepcache:
+            if self.succ_rpm_trans:
```

Comment:
hm, so what about renaming `self.succ_rpm_trans` to something like `self.clean_tempfiles` so that everybody will know what the flag does and why it shouldn't be set to `True` if a *test* transaction succeeds?